numpy/ma/core.py

```diff
@@ -3183,17 +3186,19 @@ def __setitem__(self, indx, value):
         locations.
 
         """
-        # 2016.01.15 -- v1.11.0
-        warnings.warn(
-           "Currently, slicing will try to return a view of the data," +
-           " but will return a copy of the mask. In the future, it will try" +
-           " to return both as views. This means that using `__setitem__`" +
-           " will propagate values back through all masks that are present.",
-           FutureWarning
-        )
-
         if self is masked:
             raise MaskError('Cannot alter the masked element.')
+
+        if (isinstance(self.mask, np.ndarray) and
```

Comment:
@charris this is here because I was too lazy to phase it out, I forgot the pull request, I think it was very recently by @jakirkham. This is just a thought on making the warning at least *a little* less common. I guess it is so extremely common it might almost not matter, just to note.